aptos-move/framework/aptos-framework/sources/datastructures/big_ordered_map.move

```diff
@@ -546,78 +615,29 @@ module aptos_std::big_ordered_map {
     }
 
     /// Apply the function to a reference of each element in the vector.
-    ///
-    /// Current implementation is O(n * log(n)). After function values will be optimized
-    /// to O(n).
     public inline fun for_each_ref<K: drop + copy + store, V: store>(self: &BigOrderedMap<K, V>, f: |&K, &V|) {
-        // This implementation is innefficient: O(log(n)) for next_key / borrow lookups every time,
-        // but is the only one available through the public API.
-        if (!self.is_empty()) {
-            let (k, v) = self.borrow_front();
-            f(&k, v);
-
-            let cur_k = self.next_key(&k);
-            while (cur_k.is_some()) {
-                let k = cur_k.destroy_some();
-                f(&k, self.borrow(&k));
-
-                cur_k = self.next_key(&k);
-            };
-        };
-
-        // TODO use this more efficient implementation when function values are enabled.
-        // self.for_each_leaf_node_ref(|node| {
-        //     node.children.for_each_ref(|k: &K, v: &Child<V>| {
-        //         f(k, &v.value);
-        //     });
-        // })
-    }
-
-    // TODO: Temporary friend implementaiton, until for_each_ref can be made efficient.
-    public(friend) inline fun for_each_ref_friend<K: drop + copy + store, V: store>(self: &BigOrderedMap<K, V>, f: |&K, &V|) {
         let iter = self.new_begin_iter();
```

Comment:
Unrelated to the PR, but loving that inlline functions allow us to hide that we can't capture a reference to the collection.